# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/gentoo/hotfix-bool.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Rename the bool union member, a keyword in C23.

Copyright (C) 2026 René Rebe <rene@exactco.de>
Signed-off-by: René Rebe <rene@exactco.de>

--- gentoo-0.20.7.orig/src/xmlutil.c	2014-12-28 22:06:34.000000000 +0100
+++ gentoo-0.20.7/src/xmlutil.c	2026-09-27 19:16:13.299262052 +0200
@@ -70,7 +70,7 @@ typedef union {
 	XVInt	integer;
 	XVUInt	uinteger;
 	XVReal	real;
-	XVBool	bool;
+	XVBool	boolean;
 	XVColor	color;
 } XmlData;
 
@@ -334,7 +334,7 @@ gboolean xml_get_boolean(const XmlNode *
 	if((data = get_node(node, name, XVT_BOOL)) != NULL)
 	{
 		if(x != NULL)
-			*x = data->data.bool.value;
+			*x = data->data.boolean.value;
 		return TRUE;
 	}
 	return FALSE;
@@ -677,12 +677,12 @@ static gint get_data(XmlNode *node, cons
 		node->data.type = XVT_INTEGER;
 	else if(is_word(str, XML_TRUE))
 	{
-		node->data.bool.value = TRUE;
+		node->data.boolean.value = TRUE;
 		node->data.type = XVT_BOOL;
 	}
 	else if(is_word(str, XML_FALSE))
 	{
-		node->data.bool.value = FALSE;
+		node->data.boolean.value = FALSE;
 		node->data.type = XVT_BOOL;
 	}
 	else
@@ -822,7 +822,7 @@ void xml_tree_print(const XmlNode *root,
 					printf(" = %f", root->data.real.value);
 					break;
 				case XVT_BOOL:
-					printf(" = %s", root->data.bool.value ? XML_TRUE : XML_FALSE);
+					printf(" = %s", root->data.boolean.value ? XML_TRUE : XML_FALSE);
 					break;
 				case XVT_COLOR:
 					printf(" = [%04X,%04X,%04X]", root->data.color.value.red, root->data.color.value.green, root->data.color.value.blue);
